api_test.go

```diff
@@ -337,12 +339,24 @@ func TestUpdateToStatutoryWaitingPeriod(t *testing.T) {
 		t.Run(step.name, func(t *testing.T) {
 			data, _ := os.ReadFile(step.path)
 
+			var requestData map[string]interface{}
+			if err := json.Unmarshal(data, &requestData); err != nil {
+				t.Fatalf("Failed to unmarshal JSON: %v", err)
+			}
+
+			// Log the type of 'new'
+
 			req, _ := http.NewRequest(http.MethodPost,
 				fmt.Sprintf("%s/lpas/%s/updates", baseURL, lpaUID),
 				bytes.NewReader(data))
 			withAuth(req, jwtSecretKey, authorUID)
 
 			resp, _ := doRequest(req)
+			if resp.StatusCode != http.StatusCreated {
+
+				body, _ := io.ReadAll(resp.Body)
+				t.Fatalf("Step %s failed: expected 201, got %d\nResponse body:\n%s", step.name, resp.StatusCode, string(body))
```

Comment:
I think we may want to switch this to something like:

```suggestion
                                 msg := fmt.Sprintf("Step %s failed: expected 201, got %d\nResponse body:\n%s", step.name, resp.StatusCode, string(body))
				assert.Equal(t, http.StatusCreated, resp.StatusCode, msg)
```

as `t.Fatal()` will stop the whole test suite from running (and applies to the fatyal above).